### PR TITLE
[New Rule] AWS KMS CMK Disabled or Scheduled for Deletion

### PR DESCRIPTION
**Pull request by xavigpich**
## Issues
https://github.com/elastic/detection-rules/issues/2086

## Summary
Detects an attempt to disable or schedule for deletion of an AWS KMS Customer Managed Key (CMK).

Deleting an AWS KMS key is destructive and potentially dangerous. It deletes the key material and all metadata associated with the KMS key and is irreversible. After a KMS key is deleted, you can no longer decrypt the data that was encrypted under that KMS key, which means that data becomes unrecoverable.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)? Yes

### CODE REVIEW

File: rules/integrations/aws/impact_kms_cmk_disabled_or_scheduled_for_deletion.toml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+reference = "https://attack.mitre.org/techniques/T1562/"
+[[rule.threat.technique.subtechnique]]
+id = "T1562.001"
+name = "Disable or Modify Tools"
+reference = "https://attack.mitre.org/techniques/T1562/001/"
+[rule.threat.tactic]
+id = "TA0005"
+name = "Defense Evasion"
+reference = "https://attack.mitre.org/tactics/TA0005/"
+
```
[the remark this answers is not in the document]
Comment (w0rk3r) [reply rules/integrations/aws/impact_kms_cmk_disabled_or_scheduled_for_deletion.toml:73]: I think we can remove this one

File: rules/integrations/aws/impact_kms_cmk_disabled_or_scheduled_for_deletion.toml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+name = "AWS KMS Customer Managed Key Disabled or Scheduled For Deletion"
+note = """## Setup
+
+The AWS Fleet integration, Filebeat module, or similarly structured data is required to be compatible with this rule."""
+references = [
+    "https://docs.aws.amazon.com/cli/latest/reference/kms/disable-key.html",
+    "https://docs.aws.amazon.com/cli/latest/reference/kms/schedule-key-deletion.html",
+]
+risk_score = 47
+rule_id = "x135f0du-po0z-47c7-2f6f-qsfp1x7cca9n"
```
Comment (w0rk3r) [inline rules/integrations/aws/impact_kms_cmk_disabled_or_scheduled_for_deletion.toml:37]:
```suggestion
rule_id = "6951f15e-533c-4a60-8014-a3c3ab851a1b"
```
I don't think this is a valid GUID as it contains non-hex chars

**Review by w0rk3r — CHANGES_REQUESTED** [review]
Hey @xavigpich, what a nice first contribution! I left some comments and changes, but feel free to ask for clarification if needed. Thanks!

PS: Looking forward to your next awesome contribution

**Comment by xavigpich** [comment]
Hi @w0rk3r , I appreciate your positive feedback. I've updated the PR with your suggestions. Thanks!

File: rules/integrations/aws/impact_kms_cmk_disabled_or_scheduled_for_deletion.toml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+description = """
+Identifies attempts to disable or schedule the deletion of an AWS KMS Customer Managed Key (CMK). Deleting an AWS KMS
+key is destructive and potentially dangerous. It deletes the key material and all metadata associated with the KMS key
+and is irreversible. After a KMS key is deleted, the data that was encrypted under that KMS key can no longer be
+decrypted, which means that data becomes unrecoverable.
+"""
+false_positives = [
+    """
+    A KMS customer managed key may be disabled or scheduled for deletion by a system administrator. Verify whether the user identity, user agent, 
+    and/or hostname should be making changes in your environment. Log stream deletions by unfamiliar users should be investigated. If known
```
Comment (w0rk3r) [inline rules/integrations/aws/impact_kms_cmk_disabled_or_scheduled_for_deletion.toml:20]: >  Log stream deletions by unfamiliar users should be investigated.

This needs to be adjusted to this rule.

File: rules/integrations/aws/impact_kms_cmk_disabled_or_scheduled_for_deletion.toml [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    and/or hostname should be making changes in your environment. Log stream deletions by unfamiliar users should be investigated. If known 
+    behavior is causing false positives, it can be exempted from the rule.
+    """,
+]
+from = "now-60m"
+index = ["filebeat-*", "logs-aws*"]
+interval = "10m"
+language = "kuery"
+license = "Elastic License v2"
+name = "AWS KMS Customer Managed Key Disabled or Scheduled For Deletion"
```
Comment (w0rk3r) [inline rules/integrations/aws/impact_kms_cmk_disabled_or_scheduled_for_deletion.toml:29]:
```suggestion
name = "AWS KMS Customer Managed Key Disabled or Scheduled for Deletion"
```
nit

**Review by w0rk3r — APPROVED** [review]
Some nits, otherwise LGTM. Thanks!

**Comment by w0rk3r** [comment]
@xavigpich I think your AV solution deleted the [rta/bin/Installer.msi](https://github.com/elastic/detection-rules/pull/2318/files#diff-00ff656a882eef47a11bee96328a8dabce83b699c09e2454c64ec29299ba05e0) file, can you revert it?

**Comment by xavigpich** [comment]
> This is a great addition to our AWS ruleset. @xavigpich thank you for contributing! Once `installer.msi` is put back, I will approve.

The Installer.msi is back. Apologies for that

File: rules/integrations/aws/impact_kms_cmk_disabled_or_scheduled_for_deletion.toml [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    and/or hostname should be making changes in your environment. Key deletions by unfamiliar users should be investigated. If known behavior is
+    causing false positives, it can be exempted from the rule.
+    """,
+]
+from = "now-60m"
+index = ["filebeat-*", "logs-aws*"]
+interval = "10m"
+language = "kuery"
+license = "Elastic License v2"
+name = "AWS KMS Customer Managed Key Disabled or Scheduled For Deletion"
```
Comment (w0rk3r) [inline rules/integrations/aws/impact_kms_cmk_disabled_or_scheduled_for_deletion.toml:29]:
```suggestion
name = "AWS KMS Customer Managed Key Disabled or Scheduled for Deletion"
```

**Pull request merged**